File: ClientApp/src/app/models/articleMetadata.ts
export class ArticleMetadata {
  constructor(
    public articleId: string,
    public title: string,
    public time: Date,
    public category: string,
    public tags: string[],
    public draft: boolean
  ) {}

  public static newEmpty(): ArticleMetadata {
    return new ArticleMetadata('', '', new Date(), '', [], false);
  }
}

File: Models/Article/ArticleMetadata.cs
namespace Blog.Models.Article
{
    public class ArticleMetadata
    {
        public string? Title { get; set; }

        public DateTime? Time { get; set; }

        public string? Category { get; set; }

        public string[]? Tags { get; set; }

        public bool Draft { get; set; }
    }
}
